src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -115,6 +115,39 @@ public static FormattableString CreateDescription(InputParameter operationParame
             return $"{description}{(description.ToString().EndsWith(".") ? "" : ".")} Allowed values: {BuilderHelpers.EscapeXmlDocDescription(allowedValues)}";
         }
 
+        /// <summary>
+        /// This method constructs the variable name for an input parameter. If the input parameter type is an input model type,
+        /// and the input parameter name is the same as the input parameter type name, the variable name is constructed using the supplied CSharpType name. Otherwise,
+        /// it will use the input parameter name by default.
+        /// </summary>
+        /// <param name="param">The input parameter.</param>
+        /// <param name="type">The constructed CSharpType for the input parameter.</param>
+        /// <returns>A string representing the variable name for the input parameter.</returns>
+        public static string ConstructParameterVariableName(InputParameter param, CSharpType type)
+        {
+            string variableName = string.Empty;
+
+            if (param != null)
+            {
+                string paramName = param.Name;
+                variableName = paramName.ToVariableName();
+                InputType paramInputType = param.Type;
+
+                if (paramInputType is InputModelType)
+                {
+                    var paramInputTypeName = paramInputType.Name;
+
+                    if (paramName.Equals(paramInputTypeName))
+                    {
+                        variableName = !string.IsNullOrEmpty(type.Name) ? type.Name.ToVariableName() : variableName;
+                    }
+
+                }
+            }
+
+            return variableName;
```

Comment:
previously if InputParamter was null we would throw, now we silently return an empty string.  I believe we should throw since this is an invalid scenario you can't have and empty name for a parameter